### PR TITLE
Add role and permissions API

### PR DESCRIPTION
**Pull request by amoujaes1**

### CODE REVIEW

File: clearblade.md [hunk 1]
```diff
@@ -2149,3 +2152,507 @@ function myService(req, resp){
...
+~~~javascript
+    ClearBlade.init({request:req});
+    var roles = ClearBlade.Roles();
+    var callback = function (err, roleID) {
+        if (err) {
+            resp.error(roleID);
+        } 
+        resp.success(roleID);
+    }
+    roles.create({"name":"newRole","description": "This is a new role!"}, callback);
```
[the remark this answers is not in the document]
Comment (amoujaes1) [reply clearblade.md:2184]: @yashjain28 This statement is true

File: clearblade.md [hunk 2]
```diff
@@ -2149,3 +2152,507 @@ function myService(req, resp){
...
+		}
+    permissions.getPermissionsForResource("service","serviceName", callback);
+~~~
+
+# permissions.increasePermissionsForRole(roleID, permID, requestType, callback)
+
+This method increases existing permissions for roles and resources. 
+
+* @param {string} roleID - The role's unique ID 
+* @param {string} permID - The permission's ID
```
[the remark this answers is not in the document]
Comment (amoujaes1) [reply clearblade.md:2571]: - when you create a permission
- or by calling these funcs getPermissionsForRole or getPermissionsForResource
- or from the console -- but I couldn't find it there

@yashjain28

File: clearblade.md [hunk 3]
```diff
@@ -2149,3 +2152,507 @@ function myService(req, resp){
...
+			} else {
+				resp.success(data);
+			}
+		}
+    permissions.decreasePermissionsForRole("roleID","permID",1, callback);
+~~~
+
+## permissions.removePermissionsForRole(roleID, callback)
+
+This method removes permissions for a role.
```
[the remark this answers is not in the document]
Comment (amoujaes1) [reply clearblade.md:2616]: Yes

File: clearblade.md [hunk 4]
```diff
@@ -2149,3 +2152,507 @@ function myService(req, resp){
...
+			if (err) {
+				resp.error("Parse error : " + JSON.stringify(data));
+			} else {
+				resp.success(data);
+			}
+		}
+    permissions.removePermissionsForRole("<roleID>", callback);
+~~~
+
+## permissions.removePermissionsForResource(resourceType, resourceName, callback)
```
[the remark this answers is not in the document]
Comment (amoujaes1) [reply clearblade.md:2636]: Yes

File: clearblade.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (yashjain28) [reply clearblade.md:2650]: just mention that, so that it is obvious

**Pull request merged**